src/workerd/api/global-scope.c++

```diff
@@ -459,8 +459,31 @@ jsg::Promise<void> ServiceWorkerGlobalScope::test(
   return testHandler(lock, jsg::alloc<TestController>(), eh.env.addRef(lock), eh.getCtx());
 }
 
+// This promise is used to set the timeout for hibernatable websocket events. It's expected to be
+// dropped in most cases, as long as the hibernatable websocket event promise completes before it.
+kj::Promise<void> ServiceWorkerGlobalScope::eventTimeoutPromise(uint32_t timeoutMs) {
+  auto& actor = KJ_ASSERT_NONNULL(IoContext::current().getActor());
+  co_await IoContext::current().afterLimitTimeout(timeoutMs * kj::MILLISECONDS);
+  // This is the ActorFlushReason for eviction in Cloudflare's internal implementation.
+  auto evictionCode = 2;
+  actor.shutdown(evictionCode, JSG_KJ_EXCEPTION(DISCONNECTED, Error,
+    "broken.dropped; Actor exceeded event execution time and was disconnected."));
+}
+
+kj::Promise<void> ServiceWorkerGlobalScope::setHibernatableEventTimeout(kj::Promise<void> event,
+    kj::Maybe<uint32_t> eventTimeoutMs) {
+  // If we have a maximum event duration timeout set, we should prevent the actor from running
+  // for more than the user selected duration.
+  auto timeoutMs = eventTimeoutMs.orDefault((uint32_t)0);
```

Comment:
non-blocking nit: you don't actually need the `(uint32_t)` cast here.  If you're trying to get the right type, you can use the literal `0u`.